src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultRazorDocumentMappingService.cs

```diff
@@ -258,5 +252,106 @@ public override RazorLanguageKind GetLanguageKind(RazorCodeDocument codeDocument
             // Default to Razor
             return RazorLanguageKind.Razor;
         }
+
+        private bool TryMapFromProjectedDocumentRangeStrict(RazorCodeDocument codeDocument, Range projectedRange, out Range originalRange)
+        {
+            originalRange = default;
+
+            var csharpSourceText = SourceText.From(codeDocument.GetCSharpDocument().GeneratedCode);
+            var range = projectedRange;
+            var startIndex = range.Start.GetAbsoluteIndex(csharpSourceText);
+            if (!TryMapFromProjectedDocumentPosition(codeDocument, startIndex, out var hostDocumentStart, out var _))
```

Comment:
Nit: I don't think you have to declare the var for a discard variable?

Reply:
```suggestion
            if (!TryMapFromProjectedDocumentPosition(codeDocument, startIndex, out var hostDocumentStart, out _))
```